WPFHospitalEditor/HospitalMap.xaml.cs

```diff
@@ -146,25 +147,19 @@ public void appointmentSearch_Click(object sender, RoutedEventArgs e)
                 MessageBox.Show("Invalid input.");
             }
             else
-            {
-                int index = doctorsComboBox.SelectedIndex - 1;
-                int doctorId = doctorServerController.GetDoctorsByDepartment(regularExaminationDepartment).ElementAt(index).DoctorId;
-
+            {   
+                int index = doctorsComboBox.SelectedIndex;
+                DateTime startDate = DateTime.ParseExact(startDatePicker.Text + " 08:00", "MM/dd/yyyy HH:mm", null);
+                DateTime endDate = DateTime.ParseExact(endDatePicker.Text + " 16:00", "MM/dd/yyyy HH:mm", null);
```

Comment:
Should you also allow the appropriate time to be entered next to the date?
Also, if the working time of the hospital starts in 8:00 and ends in 16:00, it would be better to store that data in some variable or constant.